db/compaction/compaction_outputs.cc

```diff
@@ -532,6 +555,39 @@ Status CompactionOutputs::AddRangeDels(
           smallest_candidate =
               InternalKey(*lower_bound, tombstone.seq_, kTypeRangeDeletion);
         }
+      } else if (lower_bound_from_range_tombstone) {
+        // Range tombstone keys can be truncated at file boundaries of the files
+        // that contain them.
+        //
+        // If this lower bound is from a range tombstone key that is not
+        // truncated, i.e., it was not truncated when reading from the input
+        // files, then its sequence number and `op_type` will be
+        // kMaxSequenceNumber and kTypeRangeDeletion (see
```

Comment:
Do you know why the untruncated seqno is kMaxSequenceNumber? I was wondering, if the untruncated seqno were the tombstone seqno, whether we could set `meta.smallest = range_tombstone_lower_bound_` above this loop instead.

Reply:
I am not sure. This is how `TruncatedRangeDelIterator` is implemented. It seems that it would be correct to just use range tombstone's sequence number. I can try explore it in another PR, as changing it can affect read paths too. If we can use tombstone's seqno, then yes the smallest key of the compaction output file could be set to `range_tombstone_lower_bound_` in either cases.